var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -107,11 +109,11 @@ class Dealii(CMakePackage):
     depends_on("netcdf-cxx",       when='+netcdf+mpi')
     depends_on("oce",              when='+oce')
     depends_on("p4est",            when='+p4est+mpi')
-    depends_on("petsc+mpi",        when='@8.4.2:+petsc+mpi')
+    depends_on("petsc+mpi",        when='@8.4.2:+petsc+mpi~64bit')
     depends_on('python',           when='@8.5.0:+python')
-    depends_on("slepc",            when='@8.4.2:+slepc+petsc+mpi')
-    depends_on("petsc@:3.6.4+mpi", when='@:8.4.1+petsc+mpi')
-    depends_on("slepc@:3.6.3",     when='@:8.4.1+slepc+petsc+mpi')
+    depends_on("slepc",            when='@8.4.2:+slepc+petsc+mpi~64bit')
+    depends_on("petsc@:3.6.4+mpi", when='@:8.4.1+petsc+mpi~64bit')
+    depends_on("slepc@:3.6.3",     when='@:8.4.1+slepc+petsc+mpi~64bit')
```

Comment:
anyway, i would still merge this as-is. Once both are merged, i will adjust deal.II to use petsc+64bit and see if it compiles.